api/client/service.proto

```diff
@@ -37,10 +39,57 @@ message ExecutionResponse {
   }
 }
 
+message RequestStreamRequest {
+  uint64 amount = 1;
+}
+
+message Condition {
+  enum ConditionType {
+    EQUALS = 0;
+    LESS_THEN = 1;
+    LESS_OR_EQUAL = 2;
+    GREATER_THEN = 3;
+    GREATER_OR_EQUAL = 4;
+  }
+
+  string name = 1;
+  ConditionType type = 2;
+}
+
+message Expectation {
+  enum DataType {
+    TEXT = 0;
+    INT = 1;
+  }
+  string name = 1;
+  string input_header = 2;
+  DataType input_header_data_type = 3;
+  Condition condition = 4;
+}
+
+message RequestStreamResponse {
+  // method/path/uri/authority/content_length can all be expressed in request headers.
+  // so technically they are redundant. In case of ambiguity headers may end up being duplicated.
+  // Note that as of today, the target upstream/cluster that will be used to replay traffic to
+  // fixes the destination address/port
+  google.protobuf.StringValue method = 1;
```

Comment:
So, this is actually a `Request` specifier? If so, it should probably live in a message called something like `RequestSpecifier`, as it seems like it's somehow response related as is.